docs/decisions/0007-tagging-app.rst

```diff
@@ -19,15 +19,15 @@ Taxonomy
 
 The ``openedx_tagging`` module defines ``openedx_tagging.core.models.Taxonomy``, whose data and functionality are self-contained to the ``openedx_tagging`` app. However in Studio, we need to be able to limit access to some Taxonomy by organization, using the same "course creator" access which limits course creation for an organization to a defined set of users.
 
-So in edx-platform, we will create the ``openedx.features.tagging`` app, to contain ``models.OrgTaxonomy``. OrgTaxonomy subclasses ``openedx_tagging.core.models.Taxonomy``, employing Django's `multi-table inheritance`_ feature, which allows the base Tag class to keep foreign keys to the Taxonomy, while allowing OrgTaxonomy to store foreign keys into Studio's Organization table.
+So in edx-platform, we will create the ``openedx.features.content_tagging`` app, to contain the models and logic for linking Organization owners to Taxonomies. There's no need to subclass Taxonomy here; we can enforce access using the ``content_tagging.api``.
 
 ObjectTag
 ~~~~~~~~~
 
 Similarly, the ``openedx_tagging`` module defined ``openedx_tagging.core.models.ObjectTag``, also self-contained to the
 ``openedx_tagging`` app.
 
-But to tag content in the LMS/Studio, we create ``openedx.features.tagging.models.ContentTag``, which subclasses ``ObjectTag``, and can then reference functionality available in the platform code.
+But to tag content in the LMS/Studio, we need to enforce ``object_id`` as a CourseKey or UsageKey type. So to do this, we subclass ``ObjectTag``, and use the ``openedx.features.tagging.registry`` to register the subclass, so that it will be picked up when this tagging API creates or resyncs object tags.
```

Comment:
If `ObjectTag.object_id` is immutable (as I think it should be), then this is really just a one-time validation that needs to happen when the `ObjectTag` is first created. In that case, a subclass is probably overkill, as there is no ongoing difference in behavior for a Course ObjectTag vs. a Block ObjectTag. You just need some way to validate the `object_id` during initial creation of `ObjectTag`, but that logic doesn't need to live in openedx-learning's tagging core.

You can absolutely keep the proxy models you've made as a nice way to do that, and they can also provide a secondary benefit of checking the `object_id` format when you're loading tags. For example, if you're writing some code that is strictly dealing with XBlocks, you can choose to use `BlockObjectTag.objects.filter(...)` to load them, and `BlockObjectTag.objects.create(...)` to create a new tag, and get that extra layer of validation. But `openedx-learning` doesn't really need to know that you're doing that.

So: what I think you don't really need is the `object_tag_class` field or `cast_object_tag` behavior, since there's no ongoing difference in behavior once the ObjectTags are saved to the database.

Also, if we say that `object_id` is always an [opaque key](https://github.com/openedx/opaque-keys/tree/master/opaque_keys/edx), then `object_type` is somewhat redundant as it can be derived from the prefix of the `object_id`. If we're not using it now, I'd suggest removing it and we can always add it later if needed.

Reply:
Done: https://github.com/openedx/openedx-learning/pull/62/files#diff-f181041ce998b0b874170c0564ba6c7c7e4cc88e1a28d0f2cf928160e3c3f9d2R56-R59